src/mapping/mapping.jl

```diff
@@ -706,3 +744,133 @@ function mapping_jacobian(::Penta6_t, cnodes, ξηζ)
+"""
+    normal(ctype::AbstractEntityType, cnodes, iside, ξ)
+    normal(::TopologyStyle, ctype::AbstractEntityType, cnodes, iside, ξ)
+
+Normal vector of the `iside`th face of a cell, evaluated at position `ξ` in the face reference element.
+So for the normal vector of the face of triangle living in a 3D space, `ξ` will be 1D (because the face
+is a line, which 1D).
+
+Beware this function needs the nodes `cnodes` and the type `ctype` of the cell (and not of the face).
+
+TODO: If `iside` is positive, then the outward normal (with respect to the cell) is returned, otherwise
+the inward normal is returned.
+
+# `::isCurvilinear`
+Note that the "face" normal vector of a curve is the "direction" vector at the given extremity.
+
+# `::isVolumic`
+``n^{loc} = J^{-\\intercal} n^{ref}``
+
+"""
+function normal(ctype::AbstractEntityType, cnodes, iside, ξ)
+    normal(topology_style(ctype, cnodes), ctype, cnodes, iside, ξ)
+end
+
+function normal(::isCurvilinear, ctype::AbstractEntityType, cnodes, iside, ξ)
+    # mapping face-reference-element (here, a node) to cell-reference-element (here, a Line)
+    # Since a Line has always only two nodes, the node is necessary the `iside`-th
+    ξ_cell = get_coords(Line())[iside]
+
+    return normalize(mapping_jacobian(ctype, cnodes, ξ_cell) .* normal(shape(ctype), iside))
+end
+
+function normal(::isSurfacic, ctype::AbstractEntityType, cnodes, iside, ξ)
+    # Get cell shape and face type and shape
+    cshape = shape(ctype)
+    ftype = facetypes(ctype)[iside]
+
+    # Get face nodes
+    fnodes = [cnodes[i] for i in faces2nodes(ctype)[iside]] # @ghislainb : need better solution to index
+
+    # Get face direction vector (face Jacobian)
+    u = mapping_jacobian(ftype, fnodes, ξ)
+
+    # Get face parametrization function
+    fp = mapping_face(cshape, iside) # mapping face-ref -> cell-ref
+
+    # Compute surface jacobian
+    J = mapping_jacobian(ctype, cnodes, fp(ξ))
+
+    # Compute vector that will help orient outward normal
+    orient = mapping(ctype, cnodes, fp(ξ)) - center(ctype, cnodes)
+
+    # Normal direction
+    n = J[:, 1] × J[:, 2] × u
+
+    # Orient normal outward and normalize
+    return normalize(orient ⋅ n .* n)
+end
+
+function normal(::isVolumic, ctype::AbstractEntityType, cnodes, iside, ξ)
+    # Cell shape
+    cshape = shape(ctype)
+
+    # Face parametrization to send ξ from ref-face-element to the ref-cell-element
+    fp = mapping_face(cshape, iside) # mapping face-ref -> cell-ref
+
+    # Inverse of the Jacobian matrix (but here `y` is in the cell-reference element)
+    # Warning : do not use `mapping_inv_jacobian` which requires the knowledge of `mapping_inv` (useless here)
+    Jinv(y) = mapping_jacobian_inv(ctype, cnodes, y)
+
+    return normalize(transpose(Jinv(fp(ξ))) * normal(cshape, iside))
+end
+
+"""
+    cell_normal(ctype::AbstractEntityType, cnodes, ξ) where {T, N}
+
+Compute the cell normal vector of an entity of topology dimension equals to (d-1) in a n-D space,
```

Comment:
"(**n**-1) in a n-D space"

Reply:
alright, I will correct it in #81 to avoid re-triggering the CI & Cie